src/compiler/binder.ts

```diff
@@ -2426,7 +2431,7 @@ namespace ts {
 
         // A VariableDeclaration with a binding pattern is ES6 syntax.
         if (isBindingPattern((<VariableDeclaration>node).name)) {
-            transformFlags = TransformFlags.AssertES6;
+            transformFlags = TransformFlags.AssertES6 | TransformFlags.ContainsBindingPattern;
         }
 
         return updateTransformFlags(node, subtreeFlags, transformFlags, TransformFlags.None);

```

Comment:
Nit: I think it will be easier if it is just inline 
```ts
updateTransformFlags(node, subtreeFlags,
isBindingPattern(node.name)  ? TransformFlags.AssertES6 | TransformFlags.ContainsBindingPattern : TransformFlags.None, TransformFlags.None);
```